app/__init__.py

```diff
@@ -23,17 +23,65 @@
 
 from flask_migrate import Migrate
 
+from flask_session import Session
+
 from flask_smorest import Api
 
 from flask_sqlalchemy import SQLAlchemy
 
+import redis
+
 import yaml
 
 
 db = SQLAlchemy()
 migrate = Migrate()
 oauth = OAuth()
-cache = Cache(config={"CACHE_TYPE": "SimpleCache"})
+session = Session()
+
+# Cache settings
+cache_config_options = [
+    "CACHE_TYPE",
+    "CACHE_KEY_PREFIX",
+    "CACHE_REDIS_HOST",
+    "CACHE_REDIS_PORT",
+    "CACHE_REDIS_URL",
+    "CACHE_REDIS_PASSWORD",
+    "CACHE_DEFAULT_TIMEOUT",
+]
+cache_config = {
+    option: os.getenv(
+        f"SDB_{option}", DefaultConfiguration.__dict__.get(option))
+    for option in cache_config_options
+}
+
+
+def check_redis_connection(cache_config):
+    if (
+        "CACHE_TYPE" in cache_config
+        and cache_config["CACHE_TYPE"] == "RedisCache"
+    ):
+        try:
+            r = redis.StrictRedis(
+                host=cache_config.get("CACHE_REDIS_HOST", "localhost"),
+                port=cache_config.get("CACHE_REDIS_PORT", 6379),
+                password=cache_config.get("CACHE_REDIS_PASSWORD"),
+            )
+            r.ping()
+            return True
+        except redis.ConnectionError:
+            return False
+    return False
+
+
+if (
```

Comment:
can you move this condition above the method definition (def check_redis_connection...)?
this doesn't look good to be present between 2 method definitions